### PR TITLE
Package sifun.1.0.0

### PR DESCRIPTION
**Pull request by kkd26**
### `sifun.1.0.0`
Interpreter for SiFun (Simple Functional) Language with three different type systems (supports Higher Rank Polymorphism)

---
* Homepage: https://github.com/kkd26/SiFun
* Source repo: git+https://github.com/kkd26/SiFun
* Bug tracker: https://github.com/kkd26/SiFun/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

File: packages/sifun/sifun.1.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+synopsis:
+  "Interpreter for SiFun (Simple Functional) Language with three different type systems (supports Higher Rank Polymorphism)"
+maintainer: ["Krzysztof Druciarek <kkd26@cam.ac.uk>"]
+authors: ["Krzysztof Druciarek <kkd26@cam.ac.uk>"]
+license: "gpl-3.0"
+homepage: "https://github.com/kkd26/SiFun"
+bug-reports: "https://github.com/kkd26/SiFun/issues"
+depends: [
+  "ocaml" 
+  "ocamlfind"
```
Comment (mseri) [inline packages/sifun/sifun.1.0.0/opam:11]:
```suggestion
  "core"
```
ocamlfind is not used, core seems to be used though

File: packages/sifun/sifun.1.0.0/opam [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+authors: ["Krzysztof Druciarek <kkd26@cam.ac.uk>"]
+license: "gpl-3.0"
+homepage: "https://github.com/kkd26/SiFun"
+bug-reports: "https://github.com/kkd26/SiFun/issues"
+depends: [
+  "ocaml" 
+  "core"
+  "dune" {>= "2.9"}
+  "odoc" {with-doc}
+  "mehnir"
```
Comment (mseri) [inline packages/sifun/sifun.1.0.0/opam:14]:
```suggestion
  "menhir"
```

File: packages/sifun/sifun.1.0.0/opam [hunk 3]
```diff
@@ -0,0 +1,38 @@
+opam-version: "2.0"
+synopsis:
+  "Interpreter for SiFun (Simple Functional) Language with three different type systems (supports Higher Rank Polymorphism)"
+maintainer: ["Krzysztof Druciarek <kkd26@cam.ac.uk>"]
+authors: ["Krzysztof Druciarek <kkd26@cam.ac.uk>"]
+license: "gpl-3.0"
+homepage: "https://github.com/kkd26/SiFun"
+bug-reports: "https://github.com/kkd26/SiFun/issues"
+depends: [
+  "ocaml"
```
Comment (mseri) [inline packages/sifun/sifun.1.0.0/opam:10]:
```suggestion
  "ocaml" {>= "4.06"}
```

List.init was introduced here

**Comment by mseri** [comment]
Have you modified the tarball? It now fails the checksum validatiom:

  - sifun.1.0.0:
            https://github.com/kkd26/SiFun/archive/HMV.tar.gz (Bad checksum, expected md5=b896c58b0360affe107147342c0aa925)

**Comment by kkd26** [comment]
First of all, thanks for all the help. And yes, I have updated the tag (and the release version) today, I will publish again later.

> Have you modified the tarball? It now fails the checksum validatiom:
> 
> * sifun.1.0.0:
>   https://github.com/kkd26/SiFun/archive/HMV.tar.gz (Bad checksum, expected md5=b896c58b0360affe107147342c0aa925)

**Comment by kkd26** [comment]
It looks like the check fails because the older version of menhir is used. In my project I use menhir 2.1, should I use different version?

**Comment by mseri** [comment]
You don't need to make a new release, it is enough that you fix it on master for the next release. Packages use the opam file from this repository, not the one in the project

**Comment by kkd26** [comment]
> You don't need to make a new release, it is enough that you fix it on master for the next release. Packages use the opam file from this repository, not the one in the project

So as I understand, I still need to use opam publish when creating a different version (because the tar has changed, so a new checksum has to be calculated)?

**Comment by mseri** [comment]
What I mean is that you don't need to re-tag and create a new tarball. You can leave the old code, and fix the opam file in the development branch for the future

**Comment by mseri** [comment]
If you re-tag the code, then you invalidate the current opam file (checksums differ) and you need to re-release it (so in taht case you need to re-run opam publish)

**Comment by mseri** [comment]
Thanks! Feel free to announce the release on discuss.ocaml.lrg. We have a community category and an announcement tag for this purpose

**Pull request merged**